llvm/lib/Transforms/Vectorize/LoopVectorize.cpp

```diff
@@ -4032,14 +4016,17 @@ LoopVectorizationCostModel::computeMaxVF(ElementCount UserVF, unsigned UserIC) {
   }
 
   auto NoScalarEpilogueNeeded = [this, &UserIC](unsigned MaxVF) {
+    if (TheLoop->getExitingBlock() != TheLoop->getLoopLatch() &&
+        !Legal->hasUncountableEarlyExit())
+      return false;
```

Comment:
Could you document what this is checking? It's not entirely obvious from just looking at the condition

Reply:
Updated. Thanks!